src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -15,148 +18,198 @@ namespace Microsoft.AspNetCore.Http.RequestDelegateGenerator.StaticRouteHandlerM
 
 internal class EndpointParameter
 {
-    public EndpointParameter(Endpoint endpoint, IParameterSymbol parameter, WellKnownTypes wellKnownTypes)
+    public EndpointParameter(Endpoint endpoint, IParameterSymbol parameter, WellKnownTypes wellKnownTypes): this(endpoint, parameter.Type, parameter.Name, wellKnownTypes)
     {
-        Type = parameter.Type;
-        SymbolName = parameter.Name;
-        LookupName = parameter.Name; // Default lookup name is same as parameter name (which is a valid C# identifier).
         Ordinal = parameter.Ordinal;
-        Source = EndpointParameterSource.Unknown;
         IsOptional = parameter.IsOptional();
         DefaultValue = parameter.GetDefaultValueString();
-        IsArray = TryGetArrayElementType(parameter, out var elementType);
+        ProcessEndpointParameterSource(endpoint, parameter, parameter.GetAttributes(), wellKnownTypes);
+    }
+
+    private EndpointParameter(Endpoint endpoint, IPropertySymbol property, IParameterSymbol? parameter, WellKnownTypes wellKnownTypes) : this(endpoint, property.Type, property.Name, wellKnownTypes)
+    {
+        Ordinal = parameter?.Ordinal ?? 0;
```

Comment:
`Ordinal` isn't used in many places, but one place it's used is to get the `ParameterInfo` for `BindAsync(HttpContext, ParameterInfo)`. While the constructor parameter is a good candidate for this when it exists, the current logic is going to use the `ParameterInfo` from `del.Method.GetParameters()` rather than the constructor.

https://github.com/dotnet/aspnetcore/blob/42d14c4bab2afb8cddc1f9683aa8ff2e20a46edb/src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Emitters/EndpointParameterEmitter.cs#L239-L241

When there is no constructor parameter, we should probably use the shared [PropertyAsParameterInfo](https://github.com/dotnet/aspnetcore/blob/42d14c4bab2afb8cddc1f9683aa8ff2e20a46edb/src/Shared/PropertyAsParameterInfo.cs).

Reply:
Good spot!

> When there is no constructor parameter, we should probably use the shared [PropertyAsParameterInfo](https://github.com/dotnet/aspnetcore/blob/42d14c4bab2afb8cddc1f9683aa8ff2e20a46edb/src/Shared/PropertyAsParameterInfo.cs).

I don't agree with this part. We already derive much of the required information used in `PropertyAsParameterInfo` in the generator (default value, optionality, etc.). I'd rather re-use these in the generated code instead of calling into an API for the generated code. The `Flattern` method defined in the current implementation isn't used outside of the OpenAPI implementations that use reflection.

Overall, I'd like to avoid generating reflection-heavy code as much as possible even though I understand it can be necessary in same cases.